### ISSUE TITLE
DropDownList does not stop propagation of Esc key

### ISSUE THREAD

**Issue by VicTachev** [telerik/kendo-ui-core#7673]
### Reproduction of the problem
1. Run [this sample](https://dojo.telerik.com/eYOVuDOQ)
2. Open Dropdown
3. Try closing its popup with `Esc` key

### Current behavior
The Dialog is also closed, alongside the DropDownList popup

### Expected/desired behavior
Only the DropDownList popup should be closed

### Environment

* **Kendo UI version:** 2023.3.1114
* **Browser:** [all]
